x-pack/platform/plugins/shared/onechat/public/application/hooks/use_last_conversation_id.ts

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { useMemo } from 'react';
+import useLocalStorage from 'react-use/lib/useLocalStorage';
+import { storageKeys } from '../storage_keys';
+import { useConversationList } from './use_conversation_list';
+
+interface UseLastConversationIdParams {
+  sessionTag?: string;
+  agentId?: string;
+}
+
+export const useLastConversationId = ({
+  sessionTag,
+  agentId,
+}: UseLastConversationIdParams = {}) => {
+  const storageKey = storageKeys.getLastConversationKey(sessionTag, agentId);
+  const [storedId] = useLocalStorage<string>(storageKey);
+
+  const { conversations, isLoading } = useConversationList({ agentId });
+
+  const lastConversationId = useMemo(() => {
+    if (storedId && conversations?.find((c) => c.id === storedId)) {
+      return storedId;
+    }
+    return undefined;
+  }, [storedId, conversations]);
```

Comment:
Why do we check if the `conversationId` exists in the list of `conversations` here? I think we should allow an incorrect `converationId` to be passed and we should handle this more gracefully on the UI - right now we just show a blank screen.

If this check is false
```
    if (storedId && conversations?.find((c) => c.id === storedId)) {
      return storedId;
    }
```
and we return `undefined`, then I think we should also update the local storage to not store this conversationId anymore, wdyt?

Reply:
we check so we dont try to load a conversation that doesnt exist. yes removing this from local storage is definetely a good idea.